### DISCUSSION TITLE
DataProxy returned an error (Data transformation failed. HTTPError: HTTP Error 403: Forbidden)

### DISCUSSION THREAD

**Discussion by fruktosa** [ckan/ckan#7991] [Q&A]
I added a new dataset then when I saw the preview an error appeared, please help
![image](https://github.com/ckan/ckan/assets/67155520/11c79679-9b9f-45ed-96cc-f61e7086538e)

**Comment by amercader** [comment]
The DataProxy is an old and deprecated external service that tries to pull the data file in order to return an structured JSON response which is then used by the resource view to display a preview of the file.

The preferred alternative is to use the [DataStore](https://docs.ckan.org/en/2.10/maintaining/datastore.html#setting-up-the-datastore) to store your file contents and provide a data API, and the [DataTables](https://docs.ckan.org/en/2.10/maintaining/data-viewer.html#datatables-view) view which is the more modern replacement to the Data Explorer one shown here.

Hope this helps

**Reply by fruktosa** [reply]
I've been using datapusher, some time ago I reactivated SSL, does this affect my datapusher? because on the old dataset there was no error when previewing, but the new dataset that I added did
